src/app/Shared/Services/Report.service.tsx

```diff
@@ -101,13 +101,102 @@ export class ReportService {
     );
   }
 
+  reportJson(recording: Recording, connectUrl: string): Observable<[string, RuleEvaluation][]> {
+    if (!recording?.reportUrl) {
+      return throwError(() => new Error('No recording report URL'));
+    }
+    return this.login.getHeaders().pipe(
+      concatMap((headers) => {
+        headers.append('Accept', 'application/json');
+        return fromFetch(recording.reportUrl, {
+          method: 'GET',
+          mode: 'cors',
+          credentials: 'include',
+          headers,
+        });
+      }),
+      concatMap(async (resp) => {
+        if (resp.ok) {
+          const obj = JSON.parse(await resp.text());
```

Comment:
I think here we can refactor as this to avoid using`async`:

```typescript
return from(
  resp
    .text()
    .then(JSON.parse)
    .then((obj) => Object.entries(obj) as [string, RuleEvaluation][])
);
```